libsigopt/compute/expected_improvement.py

```diff
@@ -26,10 +27,12 @@
 @dataclass(frozen=True, slots=True)
 class PenaltyComponents:
   penalty: numpy.ndarray
-  grad_penalty: numpy.ndarray
+  grad_penalty: numpy.ndarray | None
 
 
 class ExpectedImprovement(AcquisitionFunction):
+  predictor: GaussianProcess
```

Comment:
predictor: Predictor

Reply:
`Predictor` doesn't define `append_lie_data`. As far as I could tell, only `GaussianProcess` defines this method.

Reply:
I see, that's not ideal but that's another issue